### DISCUSSION TITLE
Some scripts is not working in Tampermonkey

### DISCUSSION THREAD

**Discussion by matiffeder** [Tampermonkey/tampermonkey#1565] [Q&A]
Hi, I am working on a script to change the name of a class in CSS:

`<div class="classname">..........`

```
(function() {
    'use strict';
    const ele1 = document.getElementsByClassName('classname')[0];
    ele1.id = 'ele1ID';
    const ele2 = document.getElementById('ele1ID');
    ele2.className = 'newclassname';
})();
```
But I found it is not working in Tampermonkey but DevTools. There is no new ID added in the `div`. I am wondering which part I missed, could someone tell me that? Also, is there a way to see the log of Tampermonkey? So that I could know the errors log. Thank you! 

Tampermonkey 4.17.6161
Firefox 103

**Comment by derjanb** [comment]
What part is not working? Does it throw an error? Did the script run too early and is ele1 present at all? What is your Tampermonkey version? And what is your browser?

**Reply by matiffeder** [reply]
Thank you for replying! I did not get an error or I don't know where to find the error log. My Tampermonkey is 4.17.6161. My browser is Firefox 103. The ID didn't add when I use the script in Tampermonkey.

I think `classname` has been presented since I could use GM_addStyle for `classname` before adding an ID.
